packages/neon_framework/packages/account_repository/lib/src/account_repository.dart

```diff
@@ -247,7 +247,7 @@ class AccountRepository {
         b
           ..client = client
           ..credentials.replace(credentials)
-          ..credentials.username = response.body.ocs.data.id;
+          ..credentials.loginName = response.body.ocs.data.id;
```

Comment:
This is not correct as it is actually the username.
We should have loginName in the Credentials and username in the Account.
The app password we receive is bound to the loginName, while we need the username in different places (e.g. to generate the user ID).

Reply:
Huh, so this means our current logic is already wrong?

Reply:
I think so. I always login via username, but if you'd try to login via E-Mail it should not work as the loginName is different from the username.

Reply:
I think it is actually fine for OCS APIs because they only use the Bearer auth where the token is not connected to the loginName.
For non-OCS APIs this is not the case because we have to use Basic auth which connects loginName and token.

I am pretty sure I can actually test a fix for this because I believe I have a case where this matters and it is not working at all at the moment.

Reply:
That's good to hear.
I have a local branch that fixes this behavior but it requires #1538 to be clean.

Reply:
Huh why does it depend on that?

Reply:
I already refactored the entire username handling into a UserDetails bloc that makes use of the new storage lib.

I can change it to use the current storage although I don't see the point of it if we decide to go with the storage anyways.